### ISSUE TITLE
Give user the possibility to add own configuration providers

### ISSUE THREAD

**Issue by DeepRobin** [SpigotMC/BungeeCord#2601]
Hi I saw in the [code](https://github.com/SpigotMC/BungeeCord/blob/master/config/src/main/java/net/md_5/bungee/config/ConfigurationProvider.java), that you don't have the possibility to add your own ConfigurationProviders, because there is no add method for the HashMap providers. I think that would be great because I wanted to add a TomlConfiguration. If you do not have objections I can also make a pull request.

**Comment by Janmm14** [comment]
I'd just grab the map field it with reflection and add it that way.

**Comment by md-5** [comment]
I'm actually not sure why ConfigurationProvider is even necessary.
Just instantiating your TomlConfiguration would have exactly the same effect, no?

**Comment by DeepRobin** [comment]
@md-5 yes it has the same effect, but why exists then the ConfigurationProvider :o

Then is the get(YamlConfiguration.class) unnecessary. And then is the configuration in the same format. Then it is simple and possible to let the user decide what type of configuration he want.

What do you think? I think a class for registration these providers must be there.

What do the others think?

(Sorry, for mistakes in my spelling, I am from germany)
